src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -34,29 +34,29 @@ public static unsafe void Fill<T>(ref T refData, nuint numElements, T value)
 
                 if (sizeof(T) == 1)
                 {
-                    vector = new Vector<byte>(Unsafe.As<T, byte>(ref tmp));
+                    vector = Vector.Create(Unsafe.As<T, byte>(ref tmp));
```

Comment:
these should rather use `Vector.LoadUnsafe(ref tmp).AsByte()` or similar.

We should be explicit that its a load API and not do extra or other things that the JIT has to elide or reason about

Reply:
Ah nevermind, I see the issue here. This is doing a broadcast and is reinterpreting the ref to a supported `T`

This pattern in general is not great and something that we probably should have a centralized helper or some other API around...

Reply:
@xtqqczze could you please introduce such a helper (internal)? so it won't look like replacing one unsafe code with another

Reply:
> we probably should have a centralized helper or some other API around...

Is there another place where this helper can be used? It looks very specific to implementation of this method.

> it won't look like replacing one unsafe code with another

This is trying to replace unsafe code with more correct and less dangerous unsafe code. I think this should just use Unsafe.BitCast: #116221.

If we factor this out to a helper, the helper is going to be as unsafe as Unsafe.BitCast, so still unsafe.